packages/buffetjs-core/src/components/DatePicker/index.js

```diff
@@ -38,13 +43,23 @@ class DatePicker extends React.PureComponent {
     this.setState({ date, displayedDate });
   }
 
+  getVisibleDate = () => {
+    const { date } = this.state;
+    const month = moment(date).month() + 1;
+    const year = moment(date).year();
+
+    return moment(`${month} ${year}`, 'MM YYYY');
+  };
+
   handleChange = ({ target }) => {
     this.setState({ displayedDate: target.value });
 
     clearTimeout(this.timer);
 
     this.timer = setTimeout(() => {
+      this.setState({ isFocused: false });
```

Comment:
Why do we need two subsequent `setState`? Can you add some comments?

Reply:
I just remove this, changing the datepicker to a functional component.

You said maybe it can be better to close the panel while the user is changing the text input and reopen it later with the right panel. I think it's the best way to avoid multiple compo updates/renders each time the user changes manually a number.